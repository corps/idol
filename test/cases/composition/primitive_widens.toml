[ReferenceRemainsRight]
is_a = ["int", "IntAlias"]

[Widens]
is_a = ["string", "int"]
variance = "Contravariant"

[AnyAlias]
is_a = ["any", "BiggerWidens"]

[BiggerWidens]
is_a = ["string", "int", "IntAlias"]
variance = "Contravariant"

[StaysNarrow]
is_a = ["string", "AnyAlias"]
variance = "Covariant"

[IntAlias]
is_a = "int"

[ReferenceRemains]
is_a = ["IntAlias", "int"]
