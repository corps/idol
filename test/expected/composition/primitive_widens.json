{
  "AnyAlias": {
    "enum": [],
    "fields": null,
    "is_a": [
      "primitive_widens.BiggerWidens"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "BiggerWidens": {
    "enum": [],
    "fields": null,
    "is_a": [
      "any"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "IntAlias": {
    "enum": [],
    "fields": null,
    "is_a": [
      "int"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "ReferenceRemains": {
    "enum": [],
    "fields": null,
    "is_a": [
      "primitive_widens.IntAlias"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "ReferenceRemainsRight": {
    "enum": [],
    "fields": null,
    "is_a": [
      "int"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "StaysNarrow": {
    "enum": [],
    "fields": null,
    "is_a": [
      "string"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  },
  "Widens": {
    "enum": [],
    "fields": null,
    "is_a": [
      "any"
    ],
    "tags": [],
    "trim": false,
    "variance": "Covariant"
  }
}
